dao/src/main/java/org/thingsboard/server/dao/cloud/DefaultEdgeSettingsService.java

```diff
@@ -0,0 +1,90 @@
+
+import com.google.common.util.concurrent.ListenableFuture;
+import lombok.RequiredArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import org.jetbrains.annotations.Nullable;
+import org.springframework.stereotype.Service;
+import org.thingsboard.common.util.JacksonUtil;
+import org.thingsboard.server.common.data.AttributeScope;
+import org.thingsboard.server.common.data.DataConstants;
+import org.thingsboard.server.common.data.edge.EdgeSettings;
+import org.thingsboard.server.common.data.id.TenantId;
+import org.thingsboard.server.common.data.kv.AttributeKvEntry;
+import org.thingsboard.server.common.data.kv.BaseAttributeKvEntry;
+import org.thingsboard.server.common.data.kv.StringDataEntry;
+import org.thingsboard.server.dao.attributes.AttributesService;
+
+import java.util.Collections;
+import java.util.List;
+import java.util.Optional;
+import java.util.concurrent.ExecutionException;
+
+import static org.thingsboard.server.dao.cloud.MessageConstants.FETCHING_EDGE_SETTINGS_ERROR_MESSAGE;
+import static org.thingsboard.server.dao.cloud.MessageConstants.FOUND_CURRENT_EDGE_SETTINGS_MESSAGE;
+import static org.thingsboard.server.dao.cloud.MessageConstants.NOT_FOUND_EDGE_SETTINGS_MESSAGE;
+import static org.thingsboard.server.dao.cloud.MessageConstants.SAVE_EDGE_SETTINGS_ERROR_MESSAGE;
+
+@Service
+@Slf4j
+@RequiredArgsConstructor
+public class DefaultEdgeSettingsService implements EdgeSettingsService {
+    public final AttributesService attributesService;
+
+    @Override
+    public EdgeSettings findEdgeSettings(TenantId tenantId) {
+        try {
+            return tryFindEdgeSettings(tenantId);
+        } catch (Exception e) {
+            log.error(FETCHING_EDGE_SETTINGS_ERROR_MESSAGE, e);
+            throw new RuntimeException(FETCHING_EDGE_SETTINGS_ERROR_MESSAGE, e);
+        }
+    }
+
+    @Nullable
+    private EdgeSettings tryFindEdgeSettings(TenantId tenantId) throws InterruptedException, ExecutionException {
+        Optional<AttributeKvEntry> attr =
+                attributesService.find(tenantId, tenantId, AttributeScope.SERVER_SCOPE, DataConstants.EDGE_SETTINGS_ATTR_KEY).get();
+
+        if (attr.isPresent()) {
+            log.trace(FOUND_CURRENT_EDGE_SETTINGS_MESSAGE, attr.get().getValueAsString());
+            return JacksonUtil.fromString(attr.get().getValueAsString(), EdgeSettings.class);
+        } else {
+            log.trace(NOT_FOUND_EDGE_SETTINGS_MESSAGE);
+            return null;
+        }
+    }
+
+    @Override
+    public ListenableFuture<List<Long>> saveEdgeSettings(TenantId tenantId, EdgeSettings edgeSettings) {
+        try {
+            return trySave(tenantId, edgeSettings);
+        } catch (Exception e) {
+            log.error(SAVE_EDGE_SETTINGS_ERROR_MESSAGE, e);
+            throw new RuntimeException(SAVE_EDGE_SETTINGS_ERROR_MESSAGE, e);
+        }
+    }
+
+    private ListenableFuture<List<Long>> trySave(TenantId tenantId, EdgeSettings edgeSettings) {
```

Comment:
Yes, it was my mistake to refactor the code. :)

Reply:
I just moved it to a separate class since it has nothing to do with events.

If it’s important, I can review his work, but at this stage, I decided to leave it as it is to avoid causing errors. I have no idea why there is an exception.